webhooks-extension/pkg/endpoints/github.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoints
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	logging "github.com/tektoncd/experimental/webhooks-extension/pkg/logging"
+	"golang.org/x/xerrors"
+)
+
+// getGitHubEnterpriseCompany returns the company from a GitHub URL
+// Returns "" for non GitHub Enterprise URL example https://github.com/owner/repo
+// Returns "company" for GitHub Enterprise URL example https://github.company.com/owner/repo
+func getGitHubEnterpriseCompany(u *url.URL) string {
+	if u.Host == "github.com" {
+		return ""
+	}
+	return strings.Split(u.Host, ".")[1]
+}
+
+// getGitHubHubbubAPI returns the API URL for the GitHub PubSubHubbub API
+func getGitHubHubbubAPI(u *url.URL) string {
+	// Public GitHub PubSubHubbub API URL is "https://api.github.com/hub"
+	hubbubAPI := "https://api.github.com/hub"
+
+	// Enterprise GitHub API URL is "https://github.company.com/api/v3/hub"
+	if company := getGitHubEnterpriseCompany(u); company != "" {
+		hubbubAPI = fmt.Sprintf("https://github.%s.com/api/v3/hub", company)
```

Comment:
Implies if you use github enterprise your server api must be github.company.com/api/v3  ...... could a company not use any dns hostname they want?  Is this a guideline given by GitHub for GHE installs?  Do we need to consider alternatives?

Reply:
Thanks Duane, good catch! I've updated the code as well as the tests to catch these cases (https://github.com/tektoncd/experimental/pull/351/files#diff-e4756d2adeff9f4613bce3d5106b8b37R108). Please let me know if there's anything else I missed!